### ISSUE TITLE
make GWCS-based Spectrum1D objects support `bin_edges`

### ISSUE THREAD

**Issue by SaOgaz** [astropy/specutils#494]
When you instantiate a `Spectrum1D` with a wavelength array, and then try and call the `bin_edges` method, it gives you an error. I suspect after looking at the code that when you instantiate `Spectrum1D` this way, it doesn't even build the WCS object.

```
In [1]: from specutils import Spectrum1D                                                                                                        

In [2]: import numpy as np                                                                                                                      

In [3]: import astropy.units as u                                                                                                               

In [4]: freqs = np.arange(50) * u.GHz                                                                                                           

In [5]: flux = np.random.random((5, len(freqs))) * u.Jy                                                                                         

In [6]: spec = Spectrum1D(spectral_axis=freqs, flux=flux)                                                                                       

In [7]: spec.bin_edges                                                                                                                          
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-413064155989> in <module>
----> 1 spec.bin_edges

~/miniconda3/envs/specutilsdev/lib/python3.7/site-packages/astropy/utils/decorators.py in __get__(self, obj, owner)
    742                 return val
    743             else:
--> 744                 val = self.fget(obj)
    745                 obj.__dict__[self._key] = val
    746                 return val

~/specutils/specutils/spectra/spectrum1d.py in bin_edges(self)
    148     @lazyproperty
    149     def bin_edges(self):
--> 150         return self.wcs.bin_edges()
    151 
    152     @property



```

**Comment by SaOgaz** [comment]
Whoops, forgot to tag @nmearl.

**Comment by nmearl** [comment]
I should note that this has nothing to do with using the wavelength property, or initializing the spectral axis in wavelength space. This is because the gwcs adapter does not provide a bin edges calculation, unlike the fits wcs adapter.

**Comment by eteq** [comment]
Hmm, so should we re-label this issue "make GWCS-based Spectrum1D objects support `bin_edges`", @nmearl?

**Comment by nmearl** [comment]
Yeah, I think that's more appropriate.

**Comment by nmearl** [comment]
This bug has been fixed in #645. Further discussion in other linked tickets.
